tools/stix-to-ecs/stix_to_ecs.py

```diff
@@ -1,588 +1,656 @@
+    :return: The list of found Json files or [path] if path is a file.
+    """
+
+    return (
+        [path]
+        if path.is_file()
+        else list(path.rglob("*.json") if recursive else path.glob("*.json"))
+    )
+
+
+def get_password() -> str:
+    """
+    Get password from user input.
+    :return: The api key entered by the user.
+    """
+
+    return getpass.getpass(
+        "Please enter your api key to connect to the Elastic cluster: "
+    )
+
+
+def is_stix_indicator(stix_object: dict) -> bool:
+    """
+    The function check if a STIX object is an indicator.
+    :param stix_object: The STIX object to be checked.
+    :return: True if the STIX object is an indicator, False otherwise.
+    """
+
+    return stix_object["type"] == "indicator"
+
+
+def is_supported_ecs_indicator(ecs_indicator: dict) -> bool:
+    """
+    The function check if a STIX object is an indicator.
+    :param stix_object: The STIX object to be checked.
+    :return: True if the STIX object is an indicator, False otherwise.
+    """
+    return (
+        ecs_indicator["threat"]["indicator"]["type"]
+        not in UNSUPPORTED_ECS_INDICATOR_TYPES
+    )
+
+
+def load_stix_objects_from_file(input_path: pathlib.Path) -> list[dict]:
+    """
+    The function load STIX objects from a file.
+    :param input_path: The path of the file containing STIX objects.
+    :return: A list of STIX objects.
+    """
+
+    with input_path.open("r") as f:
+        objects = json.load(f).get("objects", None)
+
+    if not objects:
+        raise RuntimeError('"objects field doesn\'t exist"')
+
+    return [dict(x) for x in objects]
+
+
+def load_configuration(path: pathlib.Path) -> tuple[str | None, str | None, str | None, str | None, str | None, str]:
+    """
+    Load configuration from a given JSON file.
+    :param path: Path to the configuration file.
+    :return: Tuple containing (cloud_id, api_key, url, user, password, index).
+    """
+
+    # Load JSON configuration
+    c = json.loads(path.read_text())
+    
+    # Check for presence of either cloud_id or url
+    if c["cloud_id"] and not all(c.get(k) for k in ["api_key", "index"]):
+        raise ValueError("If cloud_id is provided, api_key and index must also be provided.")
+    elif c["url"] and not all(c.get(k) for k in ["username", "password", "index"]):
+        raise ValueError("If url is provided, username, password, and index must also be provided.")
+    elif not c["cloud_id"] and not c["url"]:
+        raise ValueError("Either cloud_id or url must be provided.")
+    
+    # Convert empty strings to None
+    return tuple(c.get(k) if c.get(k) else None for k in ["cloud_id", "api_key", "url", "username", "password","index"])
+
+def main() -> None:
+    args = parse_arguments()
+
+    files = get_json_files(args.input if args.input else args.directory, args.recursive)
+    results = process_stix_files(files, args.provider)
+
+    if not args.output and not args.elastic:

```

Comment:
I like it but maybe we should keep the original logic and add the additional parameters to it as it may be simpler, consider the following example: 
 
```Python
if not args.output and not args.elastic:
    print(json.dumps(flatten_list(results), indent=4))
    return

if args.output:
    write_ecs_files(zip(files, results), args.output)

if args.elastic:
        if args.configuration:
            cloud_id, api_key, url, username, password, index = load_configuration(args.configuration)
        else:
           cloud_id, api_key, url, username, password, index = (
               args.cloud_id,
               get_password("Give me the api key message") if args.cloud_id else None,
               args.url,
               args.username,
               get_password("Give me the password message") if args.url else None,
                args.index)

        write_ecs_to_elastic(flatten_list(results), cloud_id, api_key, url, username, password, index)
```

Reply:
In this scenario we need to rework the `get_password` to take a message as parameter that we can reuse this function to get either the api key or the password